### PR TITLE
Dhertz' keymaps

### PR DESCRIPTION
**Pull request by Dhertz**
Add my user directory and keymaps for HHKB (JP), Whitefox (ISO) and Canoe (ISO).

### CODE REVIEW

File: quantum/quantum.c [hunk 1]
```diff
@@ -27,6 +27,10 @@
 #define BREATHING_PERIOD 6
 #endif
 
+#ifndef MACRO_DELAY
```
Comment (drashna) [inline quantum/quantum.c]: Could this be pulled out? 

It isn't exactly a small change.

Comment (Dhertz) [reply quantum/quantum.c]: Yes! Sorry, I forgot I rebased this on #3092 for a fix

File: keyboards/canoe/keymaps/dhertz/keymap.c [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  [1] = LAYOUT_iso(
+      KC_GRV, KC_F1, KC_F2, KC_F3, KC_F4, KC_F5, KC_F6, KC_F7, KC_F8, KC_F9, KC_F10, KC_F10, KC_F11, KC_TRNS, ISO_COUNTRY_CODE,
+      KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+      KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, CMD_SFT_L, KC_TRNS, KC_TRNS, KC_NUBS, KC_TRNS, KC_TRNS,
+      KC_TRNS, KC_NUBS, KC_TRNS, KC_TRNS, CMD_ALT_C, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_PGUP, KC_TRNS,
+      KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_HOME, KC_PGDOWN, KC_END
+      ),
+};
+
+void matrix_scan_keymap(void) {
```
Comment (drashna) [inline keyboards/canoe/keymaps/dhertz/keymap.c]: This is a HORRIBLE idea. I'm guessing that you've noticed issues with performance?  Typing is slow, right? 

The rgb functions are very resource intensive, especially on AVR boards (like the Pro Micro) 

You want this: https://docs.qmk.fm/#/custom_quantum_functions?id=layer-change-code

But you'll need to set the rgb in the `matrix_init_user` function too.

File: keyboards/hhkb/keymaps/dhertz/keymap.c [hunk 3]
```diff
@@ -0,0 +1,85 @@
+#include "hhkb.h"
```
Comment (drashna) [inline keyboards/hhkb/keymaps/dhertz/keymap.c]: Could you replace the include with `#include QMK_KEYBOARD_H` instead?

File: keyboards/hhkb/keymaps/dhertz/rules.mk [hunk 4]
```diff
@@ -0,0 +1,6 @@
+ifndef QUANTUM_DIR
```
Comment (drashna) [inline keyboards/hhkb/keymaps/dhertz/rules.mk]: This block isn't needed

File: keyboards/whitefox/keymaps/dhertz/rules.mk [hunk 5]
```diff
@@ -0,0 +1,4 @@
+ifndef QUANTUM_DIR
```
Comment (drashna) [inline keyboards/whitefox/keymaps/dhertz/rules.mk]: This block is unneccessary

File: keyboards/whitefox/whitefox.h [hunk 6]
```diff
@@ -70,4 +70,36 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
+ * |------------------------------------------------------`    |---|
+ * |CapsLo|  A|  S|  D|  F|  G|  H|  J|  K|  L|  ;|  '|  \|    |PgU|
+ * |---------------------------------------------------------------|
+ * |Shif|   |  Z|  X|  C|  V|  B|  N|  M|  ,|  .|  /|Shift |Up |PgD|
+ * |---------------------------------------------------------------|
+ * |Ctrl|Gui |Alt |         Space    |Fn0 |Alt |Gui |  |Lef|Dow|Rig|
+ * `---------------------------------------------------------------'
+ */
+
+#define KEYMAP_ISO( \
```
Comment (drashna) [inline keyboards/whitefox/whitefox.h]: Please change this to `LAYOUT_iso`

Comment (drashna) [reply keyboards/whitefox/whitefox.h]: And, if you could, change `KEYMAP` to `LAYOUT_all` and `KEYMAP_TRUEFOX` to `LAYOUT_truefox`?

File: keyboards/whitefox/keymaps/dhertz/keymap.c [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+     * |Tab  |  Q|  W|  E|  R|  T|  Y|  U|  I|  O|  P|  [|  ]|Enter|PgU|
+     * |------------------------------------------------------`    |---|
+     * |SrCtl |  A|  S|  D|  F|  G|  H|  J|  K|  L|  ;|  '|  \|    |PgD|
+     * |---------------------------------------------------------------|
+     * |Shif|  #|  Z|  X|  C|  V|  B|  N|  M|  ,|  .|  /|Shift |Up |Hom|
+     * |---------------------------------------------------------------|
+     * |NcCtl|  Alt| CTab|        LyrSpc       |CGv|Iso|CSL|Lef|Dow|Rig|
+     * `---------------------------------------------------------------'
+     */
+    [0] = KEYMAP_ISO( \
```
Comment (drashna) [inline keyboards/whitefox/keymaps/dhertz/keymap.c]: Change these to `LAYOUT_iso`

File: keyboards/whitefox/keymaps/dhertz/keymap.c [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include "whitefox.h"
```
Comment (drashna) [inline keyboards/whitefox/keymaps/dhertz/keymap.c]: Could you replace the include with `#include QMK_KEYBOARD_H` instead?

File: users/dhertz/dhertz.c [hunk 9]
```diff
@@ -0,0 +1,102 @@
+#include "dhertz.h"
+#include "quantum.h"
```
Comment (drashna) [inline users/dhertz/dhertz.c]: Redundant.  If the dhertz.h has it inclueded, it's not included here.

Same is likely true of action.h (quantum.h includes it, IIRC)

File: users/dhertz/dhertz.h [hunk 10]
```diff
@@ -0,0 +1,26 @@
+#ifndef USERSPACE
+#define USERSPACE
+
+#include "quantum.h"
+
+#undef TAPPING_TERM
```
Comment (drashna) [inline users/dhertz/dhertz.h]: `TAPPING_TERM` should go in your config.h file, NOT here, as it won't be picked up properly. 

Like this: https://github.com/qmk/qmk_firmware/blob/master/users/drashna/config.h

**Comment by drashna** [comment]
Awesome, thanks!

**Pull request merged**